pkg/asset-manager-utils/test/RebalancingRelayer.test.ts

```diff
@@ -236,41 +226,69 @@ describe('RebalancingRelayer', function () {
-              poolId,
-              sender: sender.address,
-              recipient: recipient.address,
-              userData: exitRequest.userData,
-            });
+              expectEvent.inIndirectReceipt(await receipt.wait(), pool.interface, 'Exit', {
+                poolId,
+                sender: sender.address,
+                recipient: recipient.address,
+                userData: exitRequest.userData,
+              });
 
-            const currentSenderBalance = await pool.balanceOf(sender.address);
-            expect(currentSenderBalance.lt(previousSenderBalance)).to.be.true;
+              const currentSenderBalance = await pool.balanceOf(sender.address);
+              expect(currentSenderBalance.lt(previousSenderBalance)).to.be.true;
 
-            const currentRelayerBalance = await pool.balanceOf(relayer.address);
-            expect(currentRelayerBalance).to.be.equal(previousRelayerBalance);
-          });
+              const currentRelayerBalance = await pool.balanceOf(relayer.address);
+              expect(currentRelayerBalance).to.be.equal(previousRelayerBalance);
+            });
 
-          it('rebalances the pool', async () => {
-            const receipt = await relayer.connect(sender).exitPool(poolId, recipient.address, exitRequest);
+            it('rebalances the pool', async () => {
+              const receipt = await relayer
+                .connect(sender)
+                .exitPool(poolId, recipient.address, exitRequest, tokenIncrements);
 
-            expectEvent.inIndirectReceipt(await receipt.wait(), assetManagers[0].interface, 'Rebalanced', {
-              poolId,
-              token: tokens.first.address,
+              expectEvent.inIndirectReceipt(await receipt.wait(), assetManagers[0].interface, 'Rebalance', {
+                poolId,
+              });
+
+              expectEvent.inIndirectReceipt(await receipt.wait(), assetManagers[1].interface, 'Rebalance', {
+                poolId,
+              });
             });
+          }
 
-            expectEvent.inIndirectReceipt(await receipt.wait(), assetManagers[1].interface, 'Rebalanced', {
-              poolId,
-              token: tokens.second.address,
+          context('when pool has enough cash to process exit', () => {
+            itExitsCorrectly();
+          });
+
+          context('when pool does not have enough cash to process exit', () => {
+            sharedBeforeEach('invest funds', async () => {
+              // Config invests 95% of the pool's funds to ensure lack of cash
+              const investmentConfig = {
+                targetPercentage: fp(0.95),
+                upperCriticalPercentage: fp(0.95),
+                lowerCriticalPercentage: fp(0),
+              };
+              await pool
+                .connect(admin)
+                .setAssetManagerPoolConfig(tokens.first.address, encodeInvestmentConfig(investmentConfig));
+
+              await assetManagers[0].rebalance(poolId, true);
```

Comment:
Given that this setup is somewhat complex, let's add an `expect` here checking that `cash` is indeed lower that required